docs/DeveloperGuide.adoc

```diff
@@ -742,30 +790,282 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
 (For all use cases below, the *System* is the `AddressBook` and the *Actor* is the `user`, unless specified otherwise)
 
 [discrete]
-=== Use case: Delete person
+=== Use case: Backup and restore contacts
+
+*MSS*
+
+1.  User requests to backup contacts to a particular location
+2.  AddressBook makes a copy of the existing contacts database to the specified location
+3.  User requests to restore contacts from a particular location on the local disk
+4.  AddressBook makes a backup of the current database and overrides the existing file with
+the user's backup
++
+Use case ends
+
+*Extensions*
+
+[none]
+* 2a. The backup file cannot be read during restoration
++
+[none]
+** 2a1. AddressBook shows an error message
++
+Use case resumes at step 3
+
+* 3a. The specified location cannot be written to during backup
+** 3a1. AddressBook shows an error message
++
+Use case resumes at step 1
+
+[discrete]
+=== Use case: Add multiple phone numbers
 
 *MSS*
 
 1.  User requests to list persons
 2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+3.  User requests to add multiple phone numbers to a specific person in the list
+4.  AddressBook add multiple phone numbers of the person
+
+*Extensions*
+
+* 3a. The number added is already existed
+** 3a1. AddressBook shows an error message
 +
-Use case ends.
+Use case ends
+
+[discrete]
+=== Use case: Search contacts by multiple criteria
+
+*MSS*
+
+1.  User requests to list contacts by specifying substrings of name and/or address
+2.  AddressBook displays the matching contacts in the GUI interface (if any)
+
+*Extensions*
+
+* 2a. The list is empty
++
+Use case ends
+
+[discrete]
+=== Use case: Sort persons names in alphabetical order
+
+*MSS*
+
+1. User requests to list persons
+2. AddressBook shows a list of persons unsorted
+3. User requests to sort persons in alphabetical order
 
 *Extensions*
 
 [none]
-* 2a. The list is empty.
+* 2a. The list is sorted
 +
-Use case ends.
+Use case ends
+
+* 2b. The list is empty
++
+Use case ends
+
+[discrete]
+=== Use case: Auto-correct misspelt words
+
+*MSS*
+
+1.  User inputs a misspelt word
+2.  AddressBook identifies and autocorrects the word
++
+Use case ends
+
+[discrete]
+=== Use case: Sort person by tag
+
+*MSS*
+
+1. User requests to sort persons with a specific tag
+2. AddressBook shows a list of persons with the same tag
++
+Use case ends
+
+*Extensions*
+
+* 1a. The tag is not recognizable
+** 1a1. Addressbook requests the user to input a new tag name
++
+Steps 1a - 1a1 are repeated util the tag name entered is correct
```

Comment:
may be give an option to exit? else he is stuck in the loop :stuck_out_tongue_winking_eye:

Reply:
I changed this part, so the panel will display "Unknown tag" for this case.